src/fdo/dbus.rs

```diff
@@ -20,6 +21,18 @@ use zbus::{
 use super::msg_sender;
 use crate::{peer::Peer, peers::Peers};
 
+/// The return code of the [`start_service_by_name`] method.
+///
+/// [`start_service_by_name`]: struct.DBus.html#method.start_service_by_name
+#[repr(u32)]
+#[derive(Deserialize_repr, Serialize_repr, zbus::zvariant::Type, Debug, PartialEq, Eq)]
+pub enum StartServiceByNameReply {
```

Comment:
Oh this is very nice to make this more typed. However, I think:

* `StartServiceReply` would be a better name (if there was a `StartService` method, I bet it would have the same return value).
* This should instead be added to zbus and [the client-side API](https://docs.rs/zbus/latest/zbus/fdo/struct.DBusProxy.html#method.start_service_by_name) there should use it too. Unfortunately it would be an API break so what we can do is have `fdo::DBusProxy::start_service_by_name` still return a `u32`, provide a `TryFrom<u32> for StartServiceReply` impl and create an issue for changing the method to return `StartServiceReply` in 6.0.